qiskit/circuit/quantumcircuit.py

```diff
@@ -4807,8 +4807,10 @@ def _unroll_param_dict(
                         f" but was assigned to {len(value)} values."
                     )
                 out.update(zip(parameter, value))
-            elif isinstance(parameter, str):
+            elif isinstance(parameter, str) and self.has_parameter(parameter):
                 out[self.get_parameter(parameter)] = value
+            elif isinstance(parameter, str) and not self.has_parameter(parameter):
+                out[Parameter(parameter)] = value
```

Comment:
I assume you did this to keep the check in place for the error message when `strict=True` in the calling function. It's just a bit odd looking here because I'd expect we'd just skip above. I'd be tempted to maybe move the extras logic into this function and that would simplify the logic a bit. So add `strict` as an argument to this function and collect extras for an error message if it's set here.